runtime/cmake/platform.cmake

```diff
@@ -74,3 +74,29 @@ endif()
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1")
+
+# Hook add_library and add_executable so that we can add rules to strip debug info
+# on all targets.
+# Note: when defining a function 'foo()' any existing definition of 'foo' gets mapped to '_foo'
```

Comment:
I think this belongs in OMR.
I also suggest the default for `OMR_SEPARATE_DEBUG_INFO` should be `ON` (again, in OMR).

Reply:
The thing is that for most people having separate debug info generated during the build doesnt really make sense. If you want something like that you would set the build type to `RelWithDebInfo` which would generate executables with the debug info which would be useful for local debugging, and then when you package with cpack, it  will handle separating the debug info for you. Even if you aren't going to going to use cpack, you likely have your own method for handling handling the debug info.

The way I view it, the split debug info is realistically  an openj9 specific thing, and I would have liked to put all the cmake support in openj9, however it needs to live in omr since it needs to tie in with the ddrgen.

Reply:
Moving the 'separate debug' feature to OMR would provide consistency with the autotools build flow. As things stand, cmake builds yield executables and libraries that are not stripped, making them significantly larger than from an autotools build (images/jdk is ~1.1GB vs. ~480MB).

Maybe someday it will be worthwhile to support multiple build types, but the flavor we support first should be what openj9 had before cmake was an option.